src/storage/src/hummock/cache.rs

```diff
@@ -167,24 +232,32 @@ impl<K: PartialEq + Default, T> LruHandleTable<K, T> {
         ptr
     }
 
+    /// Insert a handle into the hash table. Return the handle of the previous value if the key
+    /// exists.
     unsafe fn insert(&mut self, hash: u64, h: *mut LruHandle<K, T>) -> *mut LruHandle<K, T> {
+        debug_assert!(!h.is_null());
+        debug_assert!(!(*h).is_in_cache());
+        (*h).set_in_cache(true);
         debug_assert!(self.list.len().is_power_of_two());
         let idx = (hash as usize) & (self.list.len() - 1);
-        let (mut prev, ptr) = self.find_pointer(idx, &(*h).key);
+        let (mut prev, ptr) = self.find_pointer(idx, (*h).get_key());
         if prev.is_null() {
             self.list[idx] = h;
         } else {
             (*prev).next_hash = h;
         }
 
         if !ptr.is_null() {
-            debug_assert!((*ptr).key.eq(&(*h).key));
+            debug_assert!((*ptr).is_same_key((*h).get_key()));
+            debug_assert!((*ptr).is_in_cache());
+            // The handle to be removed is set not in cache.
+            (*ptr).set_in_cache(false);
```

Comment:
It's not necessary to `set_in_cache` here because it's a hashtable rather than lru-list. We would remove it in `remove_cache_handle`.

Reply:
I think `set_in_cache` only reflects whether the handle is in hash table or not, so it's fine to move all `set_in_cache` to the hash table so that maintaining the `in_cache` flag gets easier. 

FYI, I have checked the current code in the main branch, and all `set_in_cache(true)` is following a `table.insert` and `set_in_cache(false)` is following a `table.remove`.

Reply:
seems OK...